lib/sycamore/sycamore/tests/integration/connectors/opensearch/test_opensearch_read.py

```diff
@@ -116,33 +130,76 @@ def test_ingest_and_read(self, setup_index, exec_mode):
         for i in range(len(doc.elements) - 1):
             assert doc.elements[i].element_index < doc.elements[i + 1].element_index
 
-    def _test_ingest_and_read_via_docid_reconstructor(self, setup_index, exec_mode, cache_dir):
+    def _test_ingest_and_read_via_docid_reconstructor(self, setup_index, os_client, cache_dir):
         """
         Validates data is readable from OpenSearch, and that we can rebuild processed Sycamore documents.
         """
 
         print(f"Using cache dir: {cache_dir}")
 
         def doc_reconstructor(index_name: str, doc_id: str) -> Document:
-            prefix = f"doc-{doc_id}"
-            found = None
-            files = os.listdir(cache_dir)
-            for f in files:
-                if f.startswith(prefix):
-                    found = f
-                    break
-            assert found, "Doc not found in cache"
             import pickle
 
-            data = pickle.load(open(f"{cache_dir}/{found}", "rb"))
+            data = pickle.load(open(f"{cache_dir}/{TestOpenSearchRead.INDEX}-{doc_id}", "rb"))
             return Document(**data)
 
-        path = str(TEST_DIR / "resources/data/pdfs/Ray.pdf")
-        context = sycamore.init(exec_mode=exec_mode)
+        def doc_to_name(doc: Document, bin: bytes) -> str:
+            return f"{TestOpenSearchRead.INDEX}-{doc.doc_id}"
+
+        context = sycamore.init(exec_mode=EXEC_LOCAL)
+        hidden = str(uuid.uuid4())
+        dicts = [
```

Comment:
Most of the other tests used much smaller Documents:
https://github.com/aryn-ai/sycamore/blob/26b14f4b0fb7d6598fa4f65b2a376d63f1f4f028/lib/sycamore/sycamore/tests/unit/test_docset.py#L104
or https://github.com/aryn-ai/sycamore/blob/26b14f4b0fb7d6598fa4f65b2a376d63f1f4f028/lib/sycamore/sycamore/tests/unit/test_materialize.py#L38 (you could list comprehension this one)

Fine to leave as is.